src/main/java/user_register/frameworks_and_drivers/WelcomeScreen.java

```diff
@@ -0,0 +1,96 @@
+package user_register.frameworks_and_drivers;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.*;
+import java.io.IOException;
+
+import user_register.application_business_rules.UserRegisterInputBoundary;
+import user_register.interface_adapters.UserRegisterController;
+import user_register.interface_adapters.UserRegisterPresenter;
+import user_register.application_business_rules.UserRepoInt;
+import user_register.application_business_rules.UserRegisterInteractor;
+import entities.*;
+
+// Frameworks/Drivers layer
+
+public class WelcomeScreen extends JFrame implements ActionListener {
+
+    /**
+     * A window with a title and a JButton.
+     */
+    public WelcomeScreen() {
+
+        JLabel title = new JLabel("Welcome Screen");
+        title.setAlignmentX(Component.CENTER_ALIGNMENT);
+
+        JButton logIn = new JButton("Log in");
+        JButton signUp = new JButton("Sign up");
+
+        JPanel buttons = new JPanel();
+        buttons.add(logIn);
+        buttons.add(signUp);
+
+        logIn.addActionListener(this);
+        signUp.addActionListener(this);
+
+        JPanel main = new JPanel();
+        main.setLayout(new BoxLayout(main, BoxLayout.Y_AXIS));
+
+        main.add(title);
+        main.add(buttons);
+        this.setContentPane(main);
+        this.pack();
+    }
+
+    /**
+     * React to a button click that results in evt.
+     */
+    public void actionPerformed(ActionEvent evt) {
+        System.out.println("Click " + evt.getActionCommand());
+        if (evt.getActionCommand().equals("Sign up")) {
+        JComponent component = (JComponent) evt.getSource();
+        Window win = SwingUtilities.getWindowAncestor(component);
+        win.dispose();
+                    // Build the main program window
+        JFrame application = new JFrame("Login Example");
+        CardLayout cardLayout = new CardLayout();
+        JPanel screens = new JPanel(cardLayout);
+        application.add(screens);
+
+//         Create the parts to plug into the Use Case+Entities engine
```

Comment:
It might be an issue that this is within a UI file rather than the main because were accessing deep inner layers within the UI. Should we instead have these values created in the main file and plugged into the WelcomeScreen method call?